Content.Server/Magic/MagicSystem.cs

```diff
@@ -0,0 +1,269 @@
+﻿using Content.Server.Construction.Conditions;
+using Content.Server.Coordinates.Helpers;
+using Content.Server.Decals;
+using Content.Server.DoAfter;
+using Content.Server.Doors.Components;
+using Content.Server.Magic.Events;
+using Content.Server.Wieldable;
+using Content.Shared.Actions;
+using Content.Shared.Actions.ActionTypes;
+using Content.Shared.Doors.Components;
+using Content.Shared.Doors.Systems;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Maps;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.Map;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Magic;
+
+public sealed class MagicSystem : EntitySystem
+{
+    [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IMapManager _mapManager = default!;
+    [Dependency] private readonly DecalSystem _decals = default!;
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly SharedDoorSystem _doorSystem = default!;
+    [Dependency] private readonly DoAfterSystem _doAfter = default!;
+
+    private readonly List<EntityUid> ActiveWalls = new ();
```

Comment:
Is this needed? Doesn't look like it's used for anything.

Reply:
Door system is used for Knock.
Do After is for learning the spells via books.

Active Walls is a list to store the active force walls to make sure they have the forcewall comp. Like in the other question about this, was following what mime invisible walls did since it was handled like that. So if it doesn't need to be like this I can remove the code about it.

Reply:
Right. This is all fixed now!